src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -10832,4 +10935,1435 @@ export class StudyViewPageStore
             this.studyIds[0] === 'heme_onc_nsclc_genie_bpc'
         );
     }
+
+    readonly filteredSamplesByDetailedCancerType = remoteData<{
+        [cancerType: string]: Sample[];
+    }>({
+        await: () => [this.selectedSamples, this.cancerStudyAsClinicalData],
+        invoke: () => {
+            let groupedSamples = this.groupSamplesByCancerType(
+                this.cancerStudyAsClinicalData.result,
+                this.selectedSamples.result!,
+                'CANCER_TYPE'
+            );
+            if (_.size(groupedSamples) === 1) {
+                groupedSamples = this.groupSamplesByCancerType(
+                    this.cancerStudyAsClinicalData.result,
+                    this.selectedSamples.result!,
+                    'CANCER_TYPE_DETAILED'
+                );
+            }
+            return Promise.resolve(groupedSamples);
+        },
+    });
+
+    public groupSamplesByCancerType(
```

Comment:
This seems like an almost duplicate of https://github.com/cBioPortal/cbioportal-frontend/blob/f593e6b6abc8604561a89ee7ee4b4a83a24ca13e/src/pages/resultsView/ResultsViewPageStore.ts#L2571

It would be great if we could define utility functions for these kind of duplicates to maximize code reuse.